client/core/trade.go

```diff
@@ -811,6 +854,10 @@ func (t *trackedTrade) tick() (assetMap, error) {
 			received += match.Match.Quantity
 			quoteReceived += calc.BaseToQuote(match.Match.Rate, match.Match.Quantity)
 
+		case t.shouldBeginFindRedemption(match):
+			log.Debugf("Ready to find counter-party redemption for match %v, order %v (%v)", match.id, t.ID(), side)
+			t.beginFindRedemption(match)
+
 		case t.isRefundable(match):
```

Comment:
Related to previous comment, it's not clear how the taker could ever get to a refund in this switch if `shouldBeginFindRedemption` catches it.

I see that `refundMatch` triggers a `beginFindRedemption` in the coin not found `Refund` failure case as we discussed it should, but with `t.shouldBeginFindRedemption` before `t.isRefundable`, I'm confused about how taker refunds can happen.

Reply:
After we walked through it, I see that the `awaitingRedemption` map entry makes `shouldBeginFindRedemption` be false.